lib/puppet/provider/parsedfile.rb

```diff
@@ -311,12 +312,6 @@ def self.prefetch_target(target)
     target_records
   end
 
-  # Is there an existing record with this name?
-  def self.record?(name)
-    return nil unless @records
-    @records.find { |r| r[:name] == name }
-  end
-
```

Comment:
This is called in https://github.com/puppetlabs/puppet/blob/945beb6ef0200b1bc7879b62131282a61fc9a935/lib/puppet/provider/parsedfile.rb#L468